src/axolotl/utils/schemas/dynamic_checkpoint.py

```diff
@@ -0,0 +1,23 @@
+"""Schema for dynamic checkpoint configuration."""
+
+from pydantic import BaseModel, Field
+
+
+class DynamicCheckpointConfig(BaseModel):
+    """Configuration for dynamic checkpoint triggering during training."""
+
+    enabled: bool = Field(
+        default=False,
+        json_schema_extra={
+            "description": "Enable dynamic checkpoint triggering during training. "
+            "Create a file '.axolotl_save_checkpoint' in output_dir to trigger. "
+        },
+    )
+    check_interval: int = Field(
+        default=100,
```

Comment:
I think a default of 100 negates the benefit of this feature depending on what is already configured as save_steps.
```suggestion
        default=10,
```